presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2144,20 +2152,14 @@ private void checkHoldsLock()
     public static class Action<T>
     {
         private final ActionType type;
-        private final T data;
+        private final Optional<T> data;
         private final HdfsContext hdfsContext;
         private final HiveIdentity identity;
 
         public Action(ActionType type, T data, HdfsContext hdfsContext, HiveIdentity identity)
         {
             this.type = requireNonNull(type, "type is null");
-            if (type == ActionType.DROP) {
-                checkArgument(data == null, "data is not null");
-            }
-            else {
-                requireNonNull(data, "data is null");
```

Comment:
- We had validation that `data != null` (unless it's DROP action). We no longer have it.

- I'm not a fan of nullable fields, but it appears that change `T data` to `Optional<T> data` is causing a lot of changes here, making the actual behavioral change invisible. I would back this out from this PR.

Reply:
I've followed review comments and then introduced Optional<T> on data

Reply:
answered there (https://github.com/prestosql/presto/pull/2692#discussion_r373675285)

Reply:
Reverted most of the changes here